radix-engine/src/engine/id_validator.rs

```diff
@@ -15,13 +15,13 @@ pub enum IdValidatorError {
 pub struct IdValidator {
     id_allocator: IdAllocator,
     bucket_ids: HashMap<BucketId, usize>,
-    proof_ids: HashMap<ProofId, BucketId>,
+    proof_ids: HashMap<ProofId, Option<BucketId>>,
```

Comment:
Unclear what Option is for. Is this to distinguish between BucketProofs and VaultProofs?

Reply:
Maybe use enum instead of Option. Would make my brain hurt less :)

Reply:
For proofs received at runtime (vault proofs) and virtual proofs, the bucket id is unknown.

Enum is good idea. Will implement it to make it explicit.